src/Juvix/Nets/Bohm.hs

```diff
@@ -288,43 +326,48 @@ fanInAux2 numFan (numOther, otherLang) level = do
   traverse_ linkAll [nodeOther1, nodeOther2, nodeFan1, nodeFan2]
   deleteRewire [numFan, numOther] [other1, other2, fanIn1, fanIn2]
 
-fanInAux0 ∷ (InfoNetwork net Lang m)
-          ⇒ Node → (Node, Primar) → m ()
-fanInAux0 numFan (numOther, otherLang) = do
-  sequentalStep
-  other1 ← newNode (Primar otherLang)
-  other2 ← newNode (Primar otherLang)
-  let nodeOther1 = RELAuxiliary0 { node       = other1
+fanInAux3 ∷ (InfoNetwork net Lang m)
+          ⇒ Node → (Node, Auxiliary3) → Int → m ()
+fanInAux3 numFan (numOther, otherLang) level = do
+  incGraphSizeStep 3
+  other1 ← newNode (Auxiliary3 otherLang)
+  other2 ← newNode (Auxiliary3 otherLang)
+  other3 ← newNode (Auxiliary3 otherLang)
+  fanIn1 ← newNode (Auxiliary2 (FanIn level))
+  fanIn2 ← newNode (Auxiliary2 (FanIn level))
+  fanIn3 ← newNode (Auxiliary2 (FanIn level))
+  let nodeOther1 = RELAuxiliary3 { node       = other1
                                  , primary    = ReLink numFan Aux1
+                                 , auxiliary1 = Link (Port Aux1 fanIn3)
+                                 , auxiliary2 = Link (Port Aux1 fanIn2)
+                                 , auxiliary3 = Link (Port Aux1 fanIn1)
                                  }
-      nodeOther2 = RELAuxiliary0 { node       = other2
+      nodeOther2 = RELAuxiliary3 { node       = other2
                                  , primary    = ReLink numFan Aux2
+                                 , auxiliary1 = Link (Port Aux2 fanIn3)
+                                 , auxiliary2 = Link (Port Aux2 fanIn2)
+                                 , auxiliary3 = Link (Port Aux2 fanIn1)
                                  }
-  traverse_ linkAll [nodeOther1, nodeOther2]
-  deleteRewire [numFan, numOther] [other1, other2]
-
-fanInAux1 ∷ (InfoNetwork net Lang m)
-          ⇒ Node → (Node, Auxiliary1) → Int → m ()
-fanInAux1 numFan (numOther, otherLang) level = do
-  incGraphSizeStep 1
-  other1 ← newNode (Auxiliary1 otherLang)
-  other2 ← newNode (Auxiliary1 otherLang)
-  fanIn1 ← newNode (Auxiliary2 (FanIn level))
-  let nodeOther1 = RELAuxiliary1 { node       = other1
-                                 , primary    = ReLink numFan Aux1
-                                 , auxiliary1 = Link (Port Aux1 fanIn1)
+      nodeFan1   = RELAuxiliary3 { node       = fanIn1
```

Comment:
Sadly this layer isn't typed, so you are allowed to create a `RELAuxiliary3`, however if you look at FanIn, you had to wrap it in an Auxiliary2. Thus this doesn't work as intended